### ISSUE TITLE
`too-few-public-methods` counts dunder methods as private

### ISSUE THREAD

**Issue by kuwv** [pylint-dev/pylint#9420]
### Bug description

Currently, pylint throws false positives of "too-few-public-methods" on dunder only classes.

This includes (but not limited to):
- Descriptors
- Callables
- Context Managers
- Metaclasses

```python
class ExampleDescriptor:
    __get__(self, obj: object, owner: Optional[Type[object]] = None) -> T:
        ...
    __set__(self, obj: object, value: T) -> None:
        ...
```

These classes use builtin dunders (magic methods) from within Python and are not considered private. However, they are not counted towards the total method count within a class by pylint.

```
built_from_dunder = ExamplePublicNewCall.__new__(attribute='an example')
```

### Command used

```shell
pylint src/<app>
```

### Pylint output

```python
too-few-public-methods: Too few public methods (1/2)
```

### Expected behavior

Warnings should not be triggered for Python capabilities that are utilizing builtin language constructs as intended.

Dunders should count towards total count of R0903.

- `__enter__` / `__exit__`
- `__get__` / `__set__`
- `__call__`
- `__new__` (singleton)

### Pylint version

```shell
pylint 2.9.3
astroid 2.6.2
Python 3.9.18 (main, Aug 24 2023, 18:16:58) 
```

### References

https://github.com/pylint-dev/pylint/issues/2710

**Comment by jacobtylerwalls** [comment]
Hi, thanks for the report, but I can't reproduce with the provided example. __get__() and __set__() are both counted as public methods. Please reopen if you can provide a reproducible case, and please also attempt to reproduce on pylint 3 or higher. Thanks!
